### DISCUSSION TITLE
How to run this on mac OS?

### DISCUSSION THREAD

**Discussion by chantlong** [microsoft/FHIR-Converter#422] [Q&A: Liquid Engine]
Reading the docs for running this on the command line and I came across the following command.

```
>.\Microsoft.Health.Fhir.Liquid.Converter.Tool.exe convert -d myTemplateDirectory -r myRootTemplate -i myInputDataFolder -o myOutputDataFolder
```

exe files do not run no mac, so how would mac users run this?

Thanks

**Comment by cksavia** [comment]
Here's what I do after cloning the repo on my M1 Mac:

1. Install **dotnet-sdk** for my system's architecture (Don't remember if I downloaded from the [website](https://dotnet.microsoft.com/en-us/download) or used [homebrew](https://formulae.brew.sh/cask/dotnet-sdk).)
2. Navigate to the Microsoft.Health.Fhir.Liquid.Converter.Tool directory `cd src/Microsoft.Health.Fhir.Liquid.Converter.Tool`
3. Publish the tool with an appropriate runtime `dotnet publish --runtime osx-x64 --configuration Release -p:PublishSingleFile=true -p:PublishTrimmed=true --self-contained true` (Note that I'm using the osx-x64 runtime rather than an arm64 runtime. That's just because I'm still using the methods I used with my old Intel Mac.)
4. Navigate to the newly-created directory `cd bin/Release/netcoreapp3.1/osx-x64`
5. Now you should be able to use the tool with commands like `./Microsoft.Health.Fhir.Liquid.Converter.Tool convert -d myTemplateDirectory -r myRootTemplate -i myInputDataFolder -o myOutputDataFolder`
